mapadroid/route/RouteManagerBase.py

```diff
@@ -573,9 +573,9 @@ def get_next_location(self, origin: str) -> Optional[Location]:
 
             # determine whether we move to the next location or the prio queue top's item
             if self.delay_after_timestamp_prio is not None \
-               and ((not self._last_round_prio.get(origin, False) or self.starve_route) and
-                    self._prio_queue and len(self._prio_queue) > 0 and
-                    self._prio_queue[0][0] < time.time()):
+                    and (self.mode == 'iv_mitm' or self.starve_route or not self._last_round_prio.get(origin, False)) \
+                    and self._prio_queue \
+                    and (self.mode == 'iv_mitm' or self._prio_queue[0][0] < time.time()):
```

Comment:
These changes are not strictly necessary as long as you have 'start_route' turned on for iv_mitm. However, it feels safer to have that first check there.

As for the 2nd change here... When we're in 'iv_mitm', the first tuple element is not a timestamp.. it's basically the index at which the mon exists in the iv list. While no one would ever have a list that is as long as current epoch, it is pointless to compare it against epoch... It's always going to be true for iv_mitm.